lib/iris/_data_manager.py

```diff
@@ -261,12 +287,26 @@ def dtype(self):
     @property
     def ndim(self):
         """The number of dimensions covered by the data being managed."""
-        return self.core_data().ndim
+        return len(self.shape)
 
     @property
     def shape(self):
         """The shape of the data being managed."""
-        return self.core_data().shape
+        if self.core_data() is None:
+            result = self._shape
+        else:
+            result = self.core_data().shape
+        return result
+
+    def is_dataless(self):
+        """Determine whether the cube is dataless.
+
+        Returns
+        -------
+        bool
+
+        """
+        return (self.core_data() is None) and (self.shape is not None)
```

Comment:
@ESadek-MO Given our axiom (and if this isn't true then something is wrong) it must always be the case that:

- `self._shape = None` and (`self._lazy_array is not None` or `self._real_array is not None`)
- `self._shape is not None` and (`self._lazy_array is None` and `self._real_array is None`)

Therefore, `is_dataless` should be defined simply as `return self._shape is not None`, right?